src/ray/protobuf/serve.proto

```diff
@@ -198,3 +198,20 @@ message DeploymentStatusInfo {
 message DeploymentStatusInfoList {
   repeated DeploymentStatusInfo deployment_status_infos = 1;
 }
+
+enum ServeApplicationStatus {
+  DEPLOYING = 0;
+  RUNNING = 1;
+  DEPLOY_FAILED = 2;
+}
+
+message ServeApplicationStatusInfo {
+  ServeApplicationStatus status = 1;
+  string message = 2;
+  string deployment_timestamp = 3;
+}
+
+message StatusInfo {
+  ServeApplicationStatusInfo app_status = 1;
+  DeploymentStatusInfoList deployment_statuses = 2;
+}
```

Comment:
- how about dropping `Serve` Prefix? the whole proto is under serve namespace anyway.
- deployment_timestamp should not be a string. it's either number for unix timestmap or datetime protobuf type.
- for `StatusInfo`, how about `StatusOverview` or `ApplicationStatus`(with the inner application status renamed to deployment status.

Reply:
1. I made the protobuf names mirror their Serve dataclass counterparts (e.g. `ServeApplicationStatus` protobuf maps to the `ServeApplicationStatus` object in `common.py`). Do you think we should remove the `Serve` prefix in both the protobufs and the dataclasses, or just the protobufs?

2. Got it, I can make that fix.

3. `StatusOverview` works for me. Renaming the inner application status to `DeploymentStatus` clashes with the existing `DeploymentStatus` that we use for individual deployment statuses.
    * Do you prefer `StatusOverview` or `ApplicationStatusOverview` (with the inner app status still called `ApplicationStatusInfo`)?

Reply:
- Remove Serve prefix in both. 
- Shorter the better. Application doesn't really convey much extra information